### ISSUE TITLE
Trying to register a realisation [...] but we already have another one locally, but it's the same path

### ISSUE THREAD

**Issue by roberth** [NixOS/nix#7495]
```
error: Trying to register a realisation of 'sha256:1b9c6991d0fe9a63637677f165669adbf3374894240e44f26ca2547e33aee596!out', but we already have another one locally.
       Local:  /nix/store/34whqq7d6s0l1j2nf6spdgq1fcsjlis1-holdings-web-ci-env
       Remote: /nix/store/34whqq7d6s0l1j2nf6spdgq1fcsjlis1-holdings-web-ci-env
```

_Originally posted by @cransom in https://github.com/NixOS/nix/issues/3898#issuecomment-1357931267_

**Comment by roberth** [comment]
@Ericson2314 presumably this is the same realisation, possibly inserted by a concurrent realisation?

https://github.com/NixOS/nix/blob/b1223e1b621ed4ad11562f0ef65c65d1c78c5e4b/src/libstore/local-store.cc#L764-L816

**Comment by Ericson2314** [comment]
Yeah this should not be an error, registering realizations should be idempotent.

Now that we have derived path syntax (#4543), we ought to make some commands to manipulate these realizations more explicitly outside the build process, which at a minimum will help us uncover bugs and write more precise functional tests.
